pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -271,29 +271,30 @@ func BuildHTTPRoutesForVirtualService(
 
 	out := make([]*route.Route, 0, len(vs.Http))
 
-	catchall := false
 	for _, http := range vs.Http {
 		if len(http.Match) == 0 {
-			if r := translateRoute(push, node, http, nil, listenPort, virtualService, serviceRegistry, gatewayNames); r != nil {
+			r := translateRoute(push, node, http, nil, listenPort, virtualService, serviceRegistry, gatewayNames)
+			if r != nil {
 				out = append(out, r)
 			}
-			catchall = true
-		} else {
-			for _, match := range http.Match {
-				if r := translateRoute(push, node, http, match, listenPort, virtualService, serviceRegistry, gatewayNames); r != nil {
+			return out, nil
```

Comment:
the old code would return an error if translateRoute returns nil. is the behavior intentionally changed here? (L300 of old code)

Reply:
It is to short circuit it, here it always return a catchall route?